elastalert/config.py

```diff
@@ -262,7 +264,17 @@ def load_modules(rule, args=None):
         raise EAException('Error initializing rule %s: %s' % (rule['name'], e))
     # Instantiate alert
     try:
-        rule['alert'] = [alert(rule) for alert in rule['alert']]
+        copied_alerts = copy.copy(rule['alert'])
+        rule['alert'] = []
+        for alert in copied_alerts:
+            if type(alert) == dict:  # For inline alert configs
+                for i, alert_conf in enumerate(alert.items()):
```

Comment:
I'm not sure if we should use enumerate here. ``len(alert)`` here should generally be just 1 and ``rule_alerts`` may contain the old style alerts, so ``rule_alerts[i]`` will not always be right. I think ``rule_alert`` should be split into two lists, for inline and old style.

``inline_alerts`` could take the form of tuples of (AlertClass, alert_config) so that we don't need to enumerate and index, we can just go through each alert and instantiate it.